File: src/Components/Haversine.js
function toRad(x) {
    return (x * Math.PI) / 180;
}
export default function Haversine(lat1,lon1,lat2,lon2){
    let dLat =  toRad(lat2 - lat1);
    let dLon = toRad(lon2 - lon1);
    lat1 = toRad(lat1);
    lat2 = toRad(lat2);
    let a = Math.pow(Math.sin(dLat / 2), 2) + Math.pow(Math.sin(dLon / 2), 2) * Math.cos(lat1) * Math.cos(lat2);
    let rad = 6371;
    let c = 2 * Math.asin(Math.sqrt(a));
    return rad*c;
}

File: Go/Haversine.go
package main

import "math"

func toRadians(num float64) float64{
	return (num*math.Pi)/180
}

func Haversine(lat1 float64, lon1 float64, lat2 float64,lon2 float64) float64{
	dLat := toRadians(lat2-lat1)
	dLon := toRadians(lon2-lon1)
	lat1 = toRadians(lat1)
	lat2 = toRadians(lat2)
	a := math.Pow(math.Sin(dLat/2),2)+math.Pow(math.Sin(dLon/2),2)*math.Cos(lat1)*math.Cos(lat2)
	rad := 6371.0
	c := 2 * math.Asin(math.Sqrt(a))
	return rad*c
}
